src/main/java/yuquiz/domain/post/repository/CustomPostRepositoryImpl.java

```diff
@@ -35,7 +41,43 @@ public Page<Post> getPosts(String keyword, Long categoryId, Pageable pageable, P
         long total = jpaQueryFactory
                 .select(post.count())
                 .from(post)
-                .where(wordContain(keyword), categoryEqual(categoryId))
+                .where(
+                        wordContain(keyword),
+                        categoryEqual(categoryId),
+                        post.studyPosts.isEmpty()
+                )
+                .fetchOne();
+
+        return new PageImpl<>(posts, pageable, total);
+    }
+
+    @Override
+    public Page<Post> getPostsByStudy(Long studyId, StudyPostType type, String keyword, Long categoryId, Pageable pageable, PostSortType sort) {
+        List<Post> posts = jpaQueryFactory
+                .select(post)
+                .from(post)
+                .join(post.studyPosts, studyPost)
+                .where(
+                        studyPost.type.eq(type),
+                        studyPost.study.id.eq(studyId),
+                        wordContain(keyword),
+                        categoryEqual(categoryId)
+                )
+                .orderBy(sort.getOrder())
+                .offset(pageable.getOffset())
+                .limit(pageable.getPageSize())
+                .fetch();
+
+        long total = jpaQueryFactory
+                .select(post.count())
+                .from(post)
+                .join(post.studyPosts, studyPost)
+                .where(
+                        studyPost.type.eq(type),
+                        studyPost.study.id.eq(studyId),
+                        wordContain(keyword),
+                        categoryEqual(categoryId)
+                )
                 .fetchOne();
```

Comment:
NullPointerException을 터트린다고 경고하네요! 어차피 post.count() 조회이기 때문에 null일 가능성은 없지만 아래와 같이 수행한다면 경고를 지울 수 있습니다! 문제되는건 아닙니다~

```suggestion
                .fetch().get(0);
```

Reply:
제가 찾아본바로는 fetch().get(0)으로 바꾸는게 더 비효율적일 수 있다고 합니다.
fetch().get(0)도 값이 없다면 OutOfBounds가 터질 수 있기도 하기 때문에
null일 가능성을 없애고 싶다면 해당 결과를 Optional로 받던지, null 확인을 따로 해주는 형식으로 처리하는게 더 효율적이라고 합니다.
근데 여기선 굳이 그렇게 처리할 필요성이 없다고 생각됩니다.

Reply:
사실 둘다 좋은 방법은 아니라고 생각합니다. 결국 예외의 가능성 자체를 지우진 못하니까요.

클린 코드를 원했다면 Optional 처리로 하는게 맞는 것 같아요. 저는 해당 경고를 지우는 용도로 말씀드린겁니다! (둘다 에러가 터질 여지는 없지만 이양이면 경고가 안뜨는 방면이 더 좋으니까요~) 예외의 가능성을 지우는 것이 목적인 피드백이 아닌 경고를 지울 수 있다는 피드백이었습니다!

Reply:
> NullPointerException을 터트린다고 경고하네요! 

아마도 이걸 말씀 하신게 fetchOne() 같은 경우에는 단일 값만 불러오기 때문에 null 발생 가능성이 있어서 그런 것 같네요!
물론, 해당 로직은 post.count() 이기 때문에 아무값도 없다면 자동으로 0을 반환하여 문제가 없지만 long이 기본 자료형이기에 주의 사항으로 인해 안정성을 보장하고자 한다면 아래와 같이 바꿀 수 있을 것 같습니다.
```suggestion
              long total = Optional.ofNullable(
                jpaQueryFactory
                        .select(post.count())
                        .from(post)
                        .where(
                                wordContain(keyword),
                                categoryEqual(categoryId),
                                post.studyPosts.isEmpty()
                        )
                        .fetchOne()
        ).orElse(0L);
```

결국엔 같은 결과가 나오지만, 주의 문구가 불편하게 느껴지신다면 이러한 방법도 있을 것 같아요!

Reply:
그렇네요
경고를 지우려면 대영님이 작성해주신 코드가 괜찮은 것 같다고 생각됩니다.
다른 검색 기능에도 동일한 방식을 적용해서 수정하도록 하겠습니다.